### PR TITLE
Remove database error thows

### PR DESCRIPTION
**Pull request by crizzitello**
Clean up more throws for the database errors.
 - Remove unused Exception Include
 - Tidy `EvidenceEditor::deleteEvidence`
 - Clean up data when appaboutToClose
 - Checks in place of throws where possible 
 - Removes two Exception Classes.

### CODE REVIEW

File: src/models/codeblock.h [hunk 1]
```diff
@@ -53,9 +52,7 @@ class Codeblock {
   QByteArray encode();
  public:
   /**
-   * @brief saveCodeblock encodes the provided codeblock, then writes that codeblock to it's
-   * filePath
-   * @throws a FileError if any issues occur while writing the file
+   * @brief saveCodeblock encodes the provided codeblock, then writes that codeblock to it's filePath
```
Comment (JoelAtDeluxe) [inline src/models/codeblock.h:55]: Thanks for cleaning up the comments

File: src/main.cpp [hunk 2]
```diff
@@ -66,11 +66,17 @@ int main(int argc, char* argv[]) {
...
+
     if (!QSystemTrayIcon::isSystemTrayAvailable()) {
       handleCLI(std::vector<std::string>(argv, argv + argc));
     }
     QApplication::setQuitOnLastWindowClosed(false);
 
+    QObject::connect(&app, &QApplication::aboutToQuit, [conn] {
+        conn->close();
+        delete conn;
+    });
```
Comment (JoelAtDeluxe) [inline src/main.cpp:62]: Nice.

File: src/db/databaseconnection.cpp [hunk 3]
```diff
@@ -386,41 +379,33 @@ QStringList DatabaseConnection::getUnappliedMigrations(const QSqlDatabase &db) {
...
-      upContent.append(line + "\n");
-    }
+  const QStringList lines = allContent.split(_newLine);
+  for (const QString &line : lines) {
+    auto lowerLine = line.trimmed().toLower();
+    if (lowerLine == _migrateUp)
+      continue;
+    else if (lowerLine == _migrateDown)
+      break;
+    upContent.append(_lineTemplate.arg(line));
```
Comment (JoelAtDeluxe) [inline src/db/databaseconnection.cpp:409]: This imposes a new requirement that migrate up must come before migrate down. That's probably fine (migrations are generated this way when using the script), but it should maybe be documented in `## Adding a db migration`

File: src/main.cpp [hunk 4]
```diff
@@ -31,6 +30,10 @@ int main(int argc, char* argv[]) {
 #ifdef Q_OS_WIN
   QCoreApplication::setOrganizationName("ashirt");
 #endif
+  if (!QSqlDatabase::isDriverAvailable(QStringLiteral("QSQLITE"))) {
```
Comment (JoelAtDeluxe) [inline src/main.cpp]: This check feels a little out of place. The DatabaseConnection class cares about the driver. Main just cares about if the  database is working. It really shouldn't have to know that the underlying database driver is SQLite. 
Perhaps we should make `QSQLITE` a static field in DatabaseConnection, or move the whole check to the databaseconnection class? Then main can ask a simpler question: Are you able to establish a connection?

**Review by JoelAtDeluxe — DISMISSED** [review]
Other than the driver check, it's all fine. I'm a little worried that we're putting too many of the errors into the terminal and not providing some kind of gui error to the user in some situations. I _think_ all of the existing error windows are preserved after this migration, but I'm not positive. It might be worth looking for dead code after the style changes are complete.

File: src/db/databaseconnection.cpp (hunk 3 above)
Comment (crizzitello) [reply src/db/databaseconnection.cpp:409]: Done

Comment (crizzitello) [reply src/db/databaseconnection.cpp:409]: Added

**Review by jkennedyvz — APPROVED** [review]

**Pull request merged**